eth/downloader/downloader.go

```diff
@@ -140,26 +152,16 @@ type Downloader struct {
 	syncStatsState       stateSyncStats
 	syncStatsLock        sync.RWMutex // Lock protecting the sync stats fields
 
+	lightchain LightChain
```

Comment:
These are downloader-owned interfaces, and they are satisfied by the `core` `Blockchain` type. They get used to describe blockchain interactions, like `HasBlockAndState`, and point toward extensibility for handling different varieties of chains like LightChain. 

We never actually use a lightchain proper, which would be just headers, and TBH I'm not 100% sure what the idea is there. As our downloader is initialized, it's lightchain field will be set to just the normal `blockchain` as a fallback-default.

Reply:
https://github.com/ethereum/go-ethereum/commit/9e5f03b6c487175cc5aa1224e5e12fd573f483a7#diff-0a2825161004a664237e6781f3ed25df